pkg/backend/filestate/store_test.go

```diff
@@ -395,3 +396,90 @@ func TestProjectReferenceStore_List(t *testing.T) {
 		})
 	}
 }
+
+func TestProjectReferenceStore_ProjectExists(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		desc string
+
+		// List of file paths relative to the storage root
+		// that should exist before ListReferences is called.
+		files []string
+
+		// Project name that should exist before ProjectExists is called.
+		projectName string
+
+		// Result that should be returned by ProjectExists.
+		exist bool
+
+		// Err that should be returned by ProjectExists.
+		err error
+	}{
+		{
+			desc:        "empty string project name",
+			files:       []string{},
+			projectName: "",
+			exist:       false,
+			err:         fmt.Errorf("empty project name"),
+		},
+		{
+			desc: "project exists",
+			files: []string{
+				".pulumi/stacks/a/foo.json",
+			},
+			projectName: "a",
+			exist:       true,
+			err:         nil,
```

Comment:
a readability suggestion that is moot once `err` is disabled, but a general best practice:
For test table fields that have the default value (zero, nil, false, empty, etc.) we can just omit them from the list to keep things simple.
```suggestion
```

Reply:
Updated https://github.com/pulumi/pulumi/pull/12798/commits/48756f68b1ec77b93f1ada9bc9dba98a5d1ad806